src/coreclr/inc/switches.h

```diff
@@ -151,8 +151,7 @@
 #define FEATURE_64BIT_ALIGNMENT
 #endif
 
-// Prefer double alignment for structs and arrays with doubles. Put arrays of doubles more agressively
-// into large object heap for performance because large object heap is 8 byte aligned
+// Prefer double alignment for structs with doubles on the stack.
 #if !defined(FEATURE_64BIT_ALIGNMENT) && !defined(HOST_64BIT)
 #define FEATURE_DOUBLE_ALIGNMENT_HINT
```

Comment:
The other usage is in

https://github.com/dotnet/runtime/blob/cbd784976ced3517e87183b03446eb786e71097e/src/coreclr/vm/classlayoutinfo.cpp#L494

which can be inlined:
`if (numR8Fields * 2 > numInstanceFields && numR8Fields >= 2)`
(only usage of ShouldAlign8)

Reply:
Yes, it can be inlined but it won't make the code easier to understand. This heuristic is duplicated in AOT compilers in method with the same name: https://github.com/dotnet/runtime/blob/1548409aedce8dafcd87f962b3cef437528d108c/src/coreclr/tools/Common/JitInterface/CorInfoImpl.cs#L2259

Reply:
My point was `FEATURE_DOUBLE_ALIGNMENT_HINT` can be replaced by  `defined(FEATURE_64BIT_ALIGNMENT) || defined(TARGET_32BIT)` since the remaining two usages have nothing to do with `double` alignment.